### PR TITLE
Hw second step in security

### PR DESCRIPTION
**Pull request by AdamMudrak**

### CODE REVIEW

**Comment by AdamMudrak** [comment]
I'd like to point out that I am pull requesting more not because I prepared a good solution, but because I don't really understand if I am doing everything correctly. Specifically,
**1. In the users_roles table ensure columns together form a composite key**  - I am not sure if this is done correctly
**2. Think about how we should add columns like is_deleted. Should we add them directly to existing scripts or there is another better way?
3. Think if is it ok to hardcode identifiers in the Liquibase scripts? What is the difference between value and valueComputed in the YML scripts?** - Where can I find example of this? I've tried, not really successfully

Aaand I am not sure that getting 403 forbidden is a good thing...

File: src/main/java/com/example/onlinebookshop/security/SecurityConfig.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+    @Bean
+    public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Exception {
+        antMatcher("/swagger-ui/**");
+        return http
+                .cors(AbstractHttpConfigurer::disable)
+                .csrf(AbstractHttpConfigurer::disable)
+                .authorizeHttpRequests(
+                        auth -> auth
+                                .requestMatchers("/auth/**", "/error")
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/onlinebookshop/security/SecurityConfig.java]:
```suggestion
                                .requestMatchers(
                                        antMatcher("/api/auth/**"),
                                        antMatcher("/swagger-ui/**"),
                                        antMatcher("/v3/api-docs/**"),
                                        antMatcher("/errors")
                                )
```

File: src/main/java/com/example/onlinebookshop/services/impl/UserDetailsServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class UserDetailsServiceImpl implements UserDetailsService {
+    private final UserRepository userRepository;
+
+    @Override
+    public UserDetails loadUserByUsername(String email) throws UsernameNotFoundException {
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/onlinebookshop/services/impl/UserDetailsServiceImpl.java]:
```suggestion
    public UserDetails loadUserByUsername(String email) {
```

File: src/main/java/com/example/onlinebookshop/services/impl/UserDetailsServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+        return userRepository.findByEmail(email).orElseThrow(() -> new UsernameNotFoundException(
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/onlinebookshop/services/impl/UserDetailsServiceImpl.java]:
```suggestion
        return userRepository.findByEmail(email).orElseThrow(() -> new EntityNotFoundException(
```

File: src/main/resources/db/changelog/changes/02-create-users-table.yaml [hunk 4]
```diff
@@ -37,3 +37,6 @@ databaseChangeLog:
               - column:
                   name: shipping_address
                   type: varchar(255)
+              - column:
```
Comment (Elena-Bruyako) [inline src/main/resources/db/changelog/changes/02-create-users-table.yaml]: Don't need to add roles here, as you create table users-roles

File: src/main/resources/db/changelog/changes/04-create-users-roles-table.yaml [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+      id: create_users_roles_table
+      author: AdamMudrak
+      changes:
+      - createTable:
+          tableName: users_roles
+          columns:
+            - column:
+                name: user_id
+                type: bigint
+                constraints:
```
Comment (Elena-Bruyako) [inline src/main/resources/db/changelog/changes/04-create-users-roles-table.yaml:12]:
```suggestion
                constraints:
                    nullable: true
                    foreignKeyName: fk_user_id
                    references: users(id)
```

File: src/main/resources/db/changelog/changes/04-create-users-roles-table.yaml [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+                  primaryKey: true
+            - column:
+                name: role_id
+                type: bigint
+                constraints:
```
Comment (Elena-Bruyako) [inline src/main/resources/db/changelog/changes/04-create-users-roles-table.yaml:20]: the same

File: src/main/resources/db/changelog/changes/README.txt [hunk 7]
```diff
@@ -0,0 +1,4 @@
+The passwords are:
```
Comment (Elena-Bruyako) [inline src/main/resources/db/changelog/changes/README.txt]: Readme file shouldn't be here

File: src/main/java/com/example/onlinebookshop/services/impl/UserDetailsServiceImpl.java [hunk 8]
```diff
@@ -0,0 +1,20 @@
+package com.example.onlinebookshop.services.impl;
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/onlinebookshop/services/impl/UserDetailsServiceImpl.java]:
```suggestion
package com.example.onlinebookshop.security;
```
replace to another package

File: src/main/java/com/example/onlinebookshop/security/SecurityConfig.java [hunk 9]
```diff
@@ -0,0 +1,49 @@
+package com.example.onlinebookshop.security;
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/onlinebookshop/security/SecurityConfig.java]:
```suggestion
package com.example.onlinebookshop.config;
```
replace to another package

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Good job! Let's improve your solution!

**Review by koval-kn — APPROVED** [review]

**Pull request merged**